en/docs/design/api-security/opa-validation/overview.md

```diff
@@ -1,33 +1,55 @@
 
-## Attaching OPA Policy
+You can offload some responsibility of making the decision to authorize or not when a consumer invokes APIs based on policies attached to APIs. Both WSO2 API Manager regular Gateway and Choreo Connect supports Open Policy Agent based request validation. Open Policy Agent comprises of a policy engine that allows users (or systems) to query policies for decisions. 
+A policy for Open Policy Agent can be though as a set of rules and policy engine evaluate these rules based on the input it receives.
 
-You can attach an OPA policy to the API that you want to enforce OPA validation. In the publisher portal, select the API that you want to configure the OPA policy and navigate to **Develop -> API Configurations -> Policies**.
+WSO2 API Manager uses this unique policy engine to evaluate the requests that the Gateway receives and decides the fate of the request.
 
-Select the gateway that the API required to be deployed in the upper right corner of the page.
+<a href="{{base_path}}/assets/img/design/security/opa/opa-policy-architecture.png">
+    <img src="{{base_path}}/assets/img/design/security/opa/opa-policy-architecture.png" alt="OPA Architecture"/>
+</a>
+
+## How to use OPA for request validation?
+
+In oder to evaluate requests with OPA, first you need to deploy OPA as a host-level daemon or a sidecar container. Once a Gateway receives a request,
+it extracts a predefined set of meta-data from the request and query OPA via HTTP/HTTPS. The OPA Policy engine evaluates the request meta-data against the configured policy and returns its validation response.
+The Gateway allows or blocks the request based on the response.
+
+!!! note
+    The Gateway node should be able to communicate with this OPA server. It can be either HTTP or HTTPS, and if HTTPS is used, make sure that you upload the certificates of the OPA server to the client trust store of the Gateway.
 
-<a href="{{base_path}}/assets/img/design/security/opa-gateway-selection.png">
-    <img src="{{base_path}}/assets/img/design/security/opa-gateway-selection.png" alt="Select gateway for OPA Policy"/>
+### Attaching OPA Policy
+Once OPA engine is deployed, you have to attach the policy to the operations that you want to enforce OPA validation.
+
+1. In the Publisher Portal, select the API that you want to configure the OPA policy and navigate to **Develop -> API Configurations -> Policies**.
+2. Select the Gateway that the API is required to be deployed in the upper right corner of the page.
+3. For this functionality, there is a policy that you can invoke for the request flow named **Validate Request with OPA Policy**. Drag and drop this policy to the request flow of your required operations. 
+4. If you want to apply this to all the operations, select the **Apply to all resources** option from the left bar.
+5. Fill the configurations as per your deployment and save the page.
+6. Create a new revision from these changes and deploy it to the Gateway.
```

Comment:
Shall we link to steps on how to do this?

Reply:
Creating a new revision is a basic flow of the APIM. I think the user who refer this docs is already aware about revision and deploy steps.